SYCL/Basic/code_location_e2e.cpp

```diff
@@ -0,0 +1,159 @@
+// RUN: %clangxx -fsycl -DNDEBUG %s -o %t.out
+// RUN: %CPU_RUN_PLACEHOLDER %t.out %CPU_CHECK_PLACEHOLDER
+// RUN: %GPU_RUN_PLACEHOLDER %t.out %GPU_CHECK_PLACEHOLDER
+// RUN: %ACC_RUN_PLACEHOLDER %t.out %ACC_CHECK_PLACEHOLDER
+
+// RUN: %clangxx -fsycl %s -o %t.out
+// RUN: %CPU_RUN_PLACEHOLDER %t.out %CPU_CHECK_PLACEHOLDER
+// RUN: %GPU_RUN_PLACEHOLDER %t.out %GPU_CHECK_PLACEHOLDER
+// RUN: %ACC_RUN_PLACEHOLDER %t.out %ACC_CHECK_PLACEHOLDER
+
+/*
+    clang++ -fsycl -DNDEBUG -o smyl.bin code_location_e2e.cpp   // <<--- NDEBUG
+   will suppress fileLocation
+
+    -DNDEBUG  <-- not debugging, meaning fileName should not be retrievable (for
+   data privacy reasons)
+
+   Between having to test against NDEBUG and test from kernel code,
+   using assert/abort type of logic is not available. This test just
+   outputs strings that it checks with FileCheck.
+
+   This test uses hardcoded file code location positions. If modified, those
+   may change. pass in any value to signal "report only" and not run tests.
+
+*/
+
+#include <sycl/sycl.hpp>
+using namespace sycl;
+
+// llvm/sycl/doc/design/DeviceLibExtensions.rst
+// Our devicelib support for <cstring> only include three memory
```

Comment:
```suggestion
// Our devicelib support for <cstring> only includes three memory
```

although I might be wrong here.